DittoHealth/src/main/java/live/ditto/health/HealthScreen.kt

```diff
@@ -0,0 +1,143 @@
+package live.ditto.health
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.gestures.Orientation
+import androidx.compose.foundation.gestures.scrollable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.Typography
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+private val HealthTypography = Typography()
+
+@Composable
+fun HealthScreen(
+    viewModel: HealthViewModel = HealthViewModel(LocalContext.current)
+) {
+    val state by viewModel.state.collectAsState()
+
+    Surface {
+        HealthScreen(state = state)
+    }
+}
+
+@Composable
+private fun HealthScreen(
+    state: HealthUiState
+) {
+    if (state.isHealthy) {
+        Healthy()
+    } else {
+        NotHealthy(state)
+    }
+}
+
+@Composable
+private fun Healthy() {
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(color = Color.Green)
+    ) {
+        Text(
+            text = "System is Healthy",
+            style = HealthTypography.displayMedium,
+            color = Color.White,
+            modifier = Modifier.align(Alignment.Center)
+        )
+    }
+}
+
+@Composable
+private fun NotHealthy(state: HealthUiState) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(color = Color.Red)
+            .padding(horizontal = 4.dp)
+    ) {
+        Text(
+            text = "Not Healthy",
+            style = HealthTypography.displayMedium,
+            color = Color.White,
+
+            )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        state.causes.forEach { cause ->
+            Cause(cause = cause)
+            Spacer(modifier = Modifier.height(16.dp))
+        }
+    }
+}
```

Comment:
Minor - I'm not sure how much effort its worth putting into a quick demo app like this, but does it make sense to usehealthy / unhealthy state from the `HealthUiState` here directly and collapse `Healthy()` and `NotHealthy(state:)` into a single composable that swaps the colors based on the health check? It seems like maybe there's some duplicated UI logic that could be simplified

Reply:
I am not following you, 
will ping you in Slack, so we can talk more about it.

Reply:
After our slack conversation, there's not really any duplication going on here, and I need to get another coffee.

Reply:
@bplattenburg and I met on Slack and we agreed on the following.
- Move the logic to resolve texts from the composable to the UiState model
- Keep the existing structure of Healthy/NotHealthy composables